core/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -197,14 +205,96 @@ public XContentBuilder innerToXContent(XContentBuilder builder, Params params) t
+                } else if (Fields.TOOK.equals(currentFieldName)) {
+                    tookInMillis = parser.intValue();
+                } else if (Fields.TIMED_OUT.equals(currentFieldName)) {
+                    timedOut = parser.booleanValue();
+                } else if (Fields.TERMINATED_EARLY.equals(currentFieldName)) {
+                    terminatedEarly = parser.booleanValue();
+                }
+                else {
+                    throwUnknownField(currentFieldName, parser.getTokenLocation());
+                }
+            } else if (token == XContentParser.Token.START_OBJECT) {
+                if (InternalSearchHits.Fields.HITS.equals(currentFieldName)) {
+                    hits = InternalSearchHits.fromXContent(parser);
+                } else if (InternalAggregations.Fields.AGGREGATIONS.equals(currentFieldName)) {
+                    // TODO parse aggregation result correctly
+                    parser.skipChildren();
+                } else if (Suggest.NAME.equals(currentFieldName)) {
+                    // TODO parse suggest section
+                    parser.skipChildren();
+                } else if (SearchProfileShardResults.PROFILE_NAME.equals(currentFieldName)) {
+                    // TODO parse "profile" section
+                    parser.skipChildren();
+                } else if (RestActions.ShardsFields._SHARDS.equals(currentFieldName)) {
+                    while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                        if (token == XContentParser.Token.FIELD_NAME) {
+                            currentFieldName = parser.currentName();
+                        } else if (token.isValue()) {
+                            if (RestActions.ShardsFields.FAILED.equals(currentFieldName)) {
+                                parser.intValue(); // we don't need it but need to consume it
+                            } else if (RestActions.ShardsFields.SUCCESSFUL.equals(currentFieldName)) {
+                                successfulShards = parser.intValue();
+                            } else if (RestActions.ShardsFields.TOTAL.equals(currentFieldName)) {
+                                totalShards = parser.intValue();
+                            } else {
+                                throwUnknownField(currentFieldName, parser.getTokenLocation());
+                            }
+                        } else if (token == XContentParser.Token.START_ARRAY) {
+                            if (RestActions.ShardsFields.FAILURES.equals(currentFieldName)) {
+                                while((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                                    failures.add(ShardSearchFailure.fromXContent(parser));
+                                }
+                            } else {
+                                throwUnknownField(currentFieldName, parser.getTokenLocation());
+                            }
+                        }
```

Comment:
Right, on that level its more like an unexpected token, we have another helper for that which I will use.